### DISCUSSION TITLE
Why Frigate deletes files after approx 1 day, but there is still available storage?

### DISCUSSION THREAD

**Discussion by meaple** [blakeblackshear/frigate#14374] [Ask A Question] [answered]
I'm running Frigate 0.14.1-f4f3cfa container LXC(Debian 12) on Proxmox 8.2.4. 32GB allocated for storage. I'm not sure why Frigate is reporting 
![image](https://github.com/user-attachments/assets/70015ea9-c3a3-446c-8638-06e6c4d1f097)
then goes to 
![image](https://github.com/user-attachments/assets/bbdd1aaf-a422-4af6-9911-fc7902b97690)
here is storage around time when frigate complains about storage
![image](https://github.com/user-attachments/assets/d76c1e5a-67de-439b-8e59-acdeff930627)

I'm trying to understand why Frigate is making space, when there is still plenty left (approx. 14GB), this behaviour was not observed on Frigate 0.13

compose.yaml 
![image](https://github.com/user-attachments/assets/97abf35e-5f4c-4ed8-883e-74b71bd970bc)

Frigate config - apologies for formatting
mqtt:
  enabled: true
  host: 192.168.1.xxx
  topic_prefix: frigate
  client_id: frigate
  user: usrusr
  password: pwdpwd
  stats_interval: 60

detectors: # <---- add detectors
  coral:
    type: edgetpu
    device: pci:0

go2rtc:
  streams:
    PorchCam_sub: rtsp://usr:pwd@192.168.1.xxx:554/cam/realmonitor?channel=1&subtype=1
    DriveCam_sub: rtsp://usr:pwd@192.168.1.xxx:554/cam/realmonitor?channel=1&subtype=1
    ShedCam_sub: rtsp://usr:pwd@192.168.1.xxx:554/cam/realmonitor?channel=1&subtype=1
    BackDoorCam_sub: rtsp://usr:pwd@192.168.1.xxx:554/cam/realmonitor?channel=1&subtype=1
    PorchCam_main: rtsp://usr:pwd@192.168.1.xxx:554/cam/realmonitor?channel=1&subtype=0
    DriveCam_main: rtsp://usr:pwd@192.168.1.xxx:554/cam/realmonitor?channel=1&subtype=0
    ShedCam_main: rtsp://usr:pwd@192.168.1.xxx:554/cam/realmonitor?channel=1&subtype=0
    BackDoorCam_main: rtsp://usr:pwd@192.168.1.xxx:554/cam/realmonitor?channel=1&subtype=0
    MilaCam_main: rtsp://192.168.1.xxx/ch0_0.h264
  webrtc:   # >>http://192.168.1.***:5000/live/webrtc/
    candidates:
      - 192.168.1.***:8555
      - stun:8555

ffmpeg:
  hwaccel_args: preset-vaapi
  input_args: preset-rtsp-restream

#logger:
#default: debug

cameras:

##Front_Door##
  Front_Door: # <------ Name the camera
    ui:
      order: 1

    ffmpeg:
      inputs:
        - path: rtsp://192.168.1.***:8554/PorchCam_sub
          roles:
            - detect
        - path: rtsp://192.168.1.***:8554/PorchCam_main
          roles:
            - record

    snapshots:
      enabled: true
      bounding_box: True
      timestamp: True
      retain:
        default: 30
    record:
      enabled: true
      events:
        pre_capture: 3
        post_capture: 3
        retain:
          default: 15

    detect:
      enabled: true # <---- disable detection until you have a working camera feed
      width: 1280 # <---- update for your camera's resolution 640
      height: 720 # <---- update for your camera's resolution 360
      fps: 7
    objects:
      track:
        - person

      filters:
        person:
          min_area: 5000
          max_area: 100000
          threshold: 0.7
          mask:
            #- 883,412,478,426,299,720,0,720,0,530,0,362,0,0,1280,0,1280,720,892,720,906,474 # Seat parked other way around
            - 876,394,0,362,0,0,1280,0,1280,720,892,720,906,474

    motion:
      mask:
        - 930,376,753,379,673,380,610,350,196,351,0,351,0,365,0,0,1280,0,1280,720,900,720
    zones:
      Front_Entrance:
        coordinates: 186,397,935,372,935,720,60,720,35,654,0,562

    review:
      alerts:
        required_zones: Front_Entrance
      detections:
        required_zones: Front_Entrance

    live:
      stream_name: PorchCam_main

##Back_Door##
  Back_Door: # <------ Name the camera
    ui:
      order: 3
    ffmpeg:

      inputs:
        - path: rtsp://192.168.1.***:8554/BackDoorCam_sub

          roles:
            - detect
        - path: rtsp://192.168.1.***:8554/BackDoorCam_main
          roles:
            - record
    snapshots:
      enabled: true
      retain:
        default: 30
    record:
      enabled: true
      events:
        pre_capture: 3
        post_capture: 3
        retain:
          default: 15

    detect:
      enabled: true # <---- disable detection until you have a working camera feed
      width: 1280  # <---- update for your camera's resolution
      height: 720 # <---- update for your camera's resolution
      fps: 7
    objects:
      track:
        - person

    motion:
      mask: 0.641,0.035,0.64,0.083,0.957,0.089,0.959,0.039

    live:
      stream_name: BackDoorCam_main

##Front_Drive##

  Front_Drive: # <------ Name the camera
    ui:
      order: 2
    ffmpeg:

      inputs:
        - path: rtsp://192.168.1.***:8554/DriveCam_sub
          roles:
            - detect
        - path: rtsp://192.168.1.***:8554/DriveCam_main
          roles:
            - record

    snapshots:
      enabled: true
      bounding_box: True
      timestamp: True
      retain:
        default: 30

    record:
      enabled: true
      events:
        pre_capture: 3
        post_capture: 3
        retain:
          default: 15

    detect:
      enabled: true # <---- disable detection until you have a working camera feed
      width: 1280 # <---- update for your camera's resolution
      height: 720 # <---- update for your camera's resolution
      fps: 7
    objects:
      track:
        - person
        - car

      filters:
        person:
          min_area: 5000
          max_area: 100000
          threshold: 0.7
          mask:
            - 571,720,598,630,109,606,74,606,0,579,0,720
            #- 482,384,511,317,792,377
            #- 0,346,804,351,936,309,983,282,1183,276,1280,187,1280,0,592,0,236,0,0,0
            - 0,346,804,351,931,350,1191,720,1165,720,1280,720,1280,187,1280,0,592,0,236,0,0,0
        car:
          #min_area: 5000
          #max_area: 100000
          threshold: 0.7
          mask:
            #- 0,720,247,720,498,395,793,345,970,354,1153,720,1280,720,1280,0,676,0,0,0
            - 0,720,247,720,498,395,767,385,804,376,839,357,911,395,1153,720,1280,720,1280,0,676,0,0,0
            - 922,439,1081,720,1028,720,594,720,234,720,357,515

    motion:
      mask:
        - 488,367,269,618,599,629,580,720,303,720,0,720,0,0,897,0,1280,0,1280,720,1165,720,1015,510,877,359,800,361

    zones:
      Guest_Drive:
        coordinates: 405,503,990,499,880,391,487,402
      Main_Drive:
        coordinates: 304,619,605,625,593,720,1008,720,888,507,397,520

    review:
      alerts:
        required_zones:
          - Guest_Drive
          - Main_Drive
      detections:
        required_zones:
          - Guest_Drive
          - Main_Drive
    
    live:
      stream_name: DriveCam_main

##Back_Shed##          
  Back_Shed: # <------ Name the camera
    ui:
      order: 4
    ffmpeg:

      inputs:
        - path: rtsp://192.168.1.***:8554/ShedCam_sub
          roles:
            - detect
        - path: rtsp://192.168.1.***:8554/ShedCam_main
          roles:
            - record
    snapshots:
      enabled: true
      retain:
        default: 30
    record:
      enabled: true
      events:
        pre_capture: 3
        post_capture: 3
        retain:
          default: 15
    detect:
      enabled: true # <---- disable detection until you have a working camera feed
      width: 1280 # <---- update for your camera's resolution
      height: 720 # <---- update for your camera's resolution
      fps: 7
    objects:
      track:
        - person

    motion:
      mask: 0.641,0.035,0.64,0.083,0.957,0.089,0.959,0.039
    live:
      stream_name: ShedCam_main

##Mila_Cam##

  Mila_Cam: # <------ Name the camera
    ui:
      order: 5
    ffmpeg:
      inputs:
        - path: rtsp://192.168.1.***:8554/MilaCam_main
version: 0.14
camera_groups:
  Outdoor:
    order: 1
    icon: LuAlignHorizontalJustifyEnd
    cameras:
      - Back_Door
      - Back_Shed
      - Front_Door
      - Front_Drive

**Comment by NickM-27** [comment]
what does the storage page in the UI show

**Reply by meaple** [reply]
this is after it was cleared
![image](https://github.com/user-attachments/assets/fe855bb5-0a88-41ee-a976-235e73324aef)
but I was checking randomly and it was in line with /media/frigate

**Reply by NickM-27** [reply]
we'll have to wait until the error is occurring to get info and see what exactly is going on

**Reply by meaple** [reply]
I've enabled debug for frigate.storage,  should I add debug for anything else?

**Reply by NickM-27** [reply]
that should be plenty

**Reply by meaple** [reply]
It didn't filled storage yet, but for unknown reason Front_Drive camera shows bandwith 3533.74 MiB/hr(others report 0), then checks for storage remaining.
![image](https://github.com/user-attachments/assets/0ff5bf1a-3320-4265-b753-4afdd0053188)
I still need to wait to see what logs printing just before clenup.

At the moment I don't understand why Front_Drive camera bandwith is reported and taken into account when it's not recording 24/7 nor reporting Person/Car. 

Below is screen of space usage from evening and after night, shownig minimal increase (also it looks like reporting of free space is matching one from log)
![image](https://github.com/user-attachments/assets/e45a403f-75ef-4cb1-886f-38cb8cbeccf1)

![image](https://github.com/user-attachments/assets/f1562773-9d6f-4535-8642-8b74fd2ba0ae)

_______________________________________________________________________________________________________________________________________
I've also noticed that there is something not quite with Storage cleanup check or reporting ... edit. Noticed that missing camera doesn't need refresh .
1 camera - usage adds up and 5 cameras listed - correctly
![image](https://github.com/user-attachments/assets/f83b726e-8e5d-4799-a482-d3bd83d9f11e)

3 cameras - usage adds up and 5 cameras listed - correctly
![image](https://github.com/user-attachments/assets/8926c333-1372-4c8e-8141-5f74612d4a66)

2 cameras - usage doesn't add up and only 4 cameras listed instead of 5
![image](https://github.com/user-attachments/assets/9ac1ce60-d024-48a6-a0ae-1d68156aec59)

**Reply by NickM-27** [reply]
Right, this is working as we would expect. The storage cleanup assumes the camera will record continuously for one hour because that is entirely possible if there is a lot of activity and we need to be able to respond to that so storage does not fill up.

**Reply by meaple** [reply]
OK, I understand that. Also I'm trying to understand what triggers or why camera is reporting bandwidth (when there is no triggerable action ie. no car or preson moving). Is this can be caused by motion ie. in car window/roof reflection ?

**Reply by NickM-27** [reply]
for your config if recordings are going to be saved then it will be because some object was detected

**Comment by meaple** [comment]
@NickM-27 This is Debug (all) snip from time around when clips are deleted. I'm wondering why it produces an Error and why it thinks I have 780 recordings expiring? Retention is set for 15, but there was no older recordings than approx 2 days.
![image](https://github.com/user-attachments/assets/417cc2c1-8e20-43ae-a68c-816ce5c15789)

**Reply by NickM-27** [reply, marked as answer]
Like it says in the logs you had less space than an hour worth of continuous recording so it had to cleanup

**Reply by meaple** [reply]
I've seen that, just curious why it was erroring and thinks that there are expiring records. Thank you for looking.

**Reply by NickM-27** [reply]
The records are expiring also meaning being deleted forcefully due to the lack of storage

**Reply by meaple** [reply]
thanks, any tip how to reduce camera bandwidth?
